### PR TITLE
add installation of different Artifactory flavours

### PR DESCRIPTION
**Pull request by weakcamel**
---
name: add installation of different Artifactory flavours
---

**Describe the change**

Introduce a parameter to select the Artifactory flavour to install.
Document the need to override download URL in case of Pro version

**Testing**

n/a yet - I lost the local changes I originally made and haven't re-tested these yet.

### CODE REVIEW

File: vars/main.yml [hunk 1]
```diff
@@ -11,8 +11,13 @@ _artifactory_requirements:
 
 artifactory_requirements: "{{ _artifactory_requirements[ansible_os_family] | default(_artifactory_requirements['default']) }}"
 
-artifactory_zip: "https://dl.bintray.com/jfrog/artifactory/jfrog-artifactory-oss-{{ artifactory_version }}.zip"
-artifactory_home: "{{ artifactory_download_directory }}/artifactory-oss-{{ artifactory_version }}"
+# Pick the Artifactory flavour to install, can be also cpp-ce, jcr, pro.
```
Comment (robertdebock) [inline vars/main.yml]: The `vars/main.yml` file is for variables used by the role, but are not "exposed" to the user. The `defaults/main.yml` file is intended for users to overwrite values.

Can you move the `artifactory_flavour` variable (and the nice description) to `defaults/main.yml`?

Comment (weakcamel) [reply vars/main.yml]: Sure, will do.

Additional question though: sice the `artifactory_zip` might need to be overridden, shall I move it as well or leave it as-is?

Comment (robertdebock) [reply vars/main.yml]: I think the `artifactory_zip` is a variable that the role can determine, based on variable set in `defaults/main.yml`, so it's in the right place in my opinion.

**Comment by robertdebock** [comment]
Test has been successful, please move `artifactory_flavour` to `defaults/main.yml` and I'm ready to merge.

File: vars/main.yml (hunk 1 above)
Comment (weakcamel) [reply vars/main.yml]: Sounds good.
Changes done.

**Comment by sm97** [comment]
@robertdebock Somehow commits from this PR did not get into master branch.

**Comment by robertdebock** [comment]
> @robertdebock Somehow commits from this PR did not get into master branch.

Hm, weird, let me review and potentially bring back that code.

Thanks for addressing the issue.

**Comment by robertdebock** [comment]
Hm, the whole structure of the role has been changed to use packages (rpm/deb) over .tar.gz.

I've just update the role (ci is testing now) to allow for "oss", "jcr" or "cpp-ce".

**Pull request merged**